### PR TITLE
Experiment adding 'how to install' instructions to /download/thanks/ (Fixes #6704)

### LINKED ISSUE #6704: [Experiment] "How To Install" content on /thanks page
Mana Page 
https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=90734715

[Current copy](https://docs.google.com/document/d/1TYsgHFCMgj7xQiveNRAf-m73KH-c9DBYCt6jMPdC-aM/edit)

| Estimates |  |
| ------------- | ------------- |
| TL | 3 days |
| PM | 1 hour |
| ENG | 5 days |
| SEO | |
| ANA | |
| DES | 1/2 day |
| COPY | 5 days |

We should add instructional content explaining to Chrome/Windows downloaders what to do next, in order to _install_ their new copy of Firefox. 

We should measure whether the download:install ratio changes for people who see this messaging. 

We should compare the install rate to two other segments: Chrome/Windows downloaders who get the current experience (a newsletter signup form) and Chrome/Windows downloaders who get a generic "thanks" message (which was the treatment prior to the newsletter signup form).

---

#### :yellow_heart: Success Criteria :yellow_heart:
- [ ] A percentage of Chrome/Windows downloaders see some clear instructions for how to install Firefox; a percentage of Chrome/Windows downloaders see the current newsletter signup form; a percentage of Chrome/Windows downloaders see a generic "thanks" message
- [ ] We can measure the install rate of these segments of downloaders and can see installs happening from each segment

#### :heavy_exclamation_mark: Risks :heavy_exclamation_mark:
- [ ] We may see that the FxA signup rate goes down

#### Tasks
- [ ]

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
Experiment parameters:

Target audience: Chrome visitors on Windows only.
Target locale: `en-US` only.
Base URL: https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/new/

There are 3 variants:

- https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/new/?v=a (control, 50%)
- https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/new/?v=b (install copy, 25%)
- https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/new/?v=c (app store buttons, 25%).

Landing on any of the variants above, I've routed around our usual stub attribution code to make a custom auth request with the following values (this avoids using same site UTM params):

```javascript
{
    utm_source: 'www.mozilla.org',
    utm_medium: 'download_button',
    utm_campaign: 'download_thanks_page',
    utm_content: 'download_thanks_install_experiment-v-{a|b|c}',
    referrer: document.referrer
}
```

Each of the 3 variants then also links directly to 4 /thanks/ page variations:

- https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/download/thanks/?v=a
- https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/download/thanks/?v=b
- https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/download/thanks/?v=c

## Issue / Bugzilla link
#6704

## Testing
- [x] Experiment should trigger for Chrome users on Windows, with DNT disabled.
- [x] Experiment should **not** trigger if a `utm_content` param is already in the /new URL.
- [ ] Any other `utm_*`params (with the exception of `utm_content`) should be passed through to the experiment variation URLs.
- [x] When clicking the download button on any page variations, it should route to the expected /thanks/ page variation.

### CODE REVIEW

**Comment by alexgibson** [comment]
Ok this is ready for code review. Happy to talk though the logic with whomever reviews this, if it helps.

File: media/js/firefox/new/install/experiment.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function(Mozilla) {
+    'use strict';
+
+    var isChrome = !!window.chrome;
```
Comment (stephaniehobson) [inline media/js/firefox/new/install/experiment.js]: The experiment is activating on Edge in windows for me. `window.chrome` returns an object.

File: bedrock/firefox/templates/firefox/new/install/scene2-b.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{% endblock %}
+
+{% block main_content %}
+<div class="install-instructions">
+  <h1 class="headline">Thanks – you’re almost done</h1>
+  <p>Click the Firefox installer in Chrome’s download bar below to finish setting up Firefox.</p>
+  <p><a href="https://support.mozilla.org/kb/how-download-and-install-firefox-windows">Need help installing Firefox?</a></p>
+</div>
+
+<img class="browser-illustration" src="{{ static('img/firefox/new/install/browser-illustration.svg') }}" alt="" height="249" width="360">
```
Comment (stephaniehobson) [inline bedrock/firefox/templates/firefox/new/install/scene2-b.html]: SVG 😍

**Comment by stephaniehobson** [comment]
How can I inspect the "custom auth request"? is that in the request URL a request header?

File: bedrock/firefox/views.py [hunk 3]
```diff
@@ -636,12 +636,15 @@ def download_thanks(request):
     locale = l10n_utils.get_locale(request)
     variant = request.GET.get('v', None)
     show_newsletter = locale in ['en-US', 'en-GB', 'en-CA', 'en-ZA', 'es-ES', 'es-AR', 'es-CL', 'es-MX', 'pt-BR', 'fr', 'ru', 'id', 'de', 'pl']
-    pre_download_locales = ['en-US', 'en-GB', 'en-CA', 'en-ZA', 'de', 'fr']
+    pre_download_locales = ['fr']
```
Comment (stephaniehobson) [inline bedrock/firefox/views.py]: What's going on here? Rebase weirdness?

File: bedrock/firefox/views.py [hunk 4]
```diff
@@ -636,12 +636,15 @@ def download_thanks(request):
...
 
     # ensure variant matches pre-defined value
-    if variant not in ['a']:  # place expected ?v= values in this list
+    if variant not in ['a', 'b', 'c', 'd']:  # place expected ?v= values in this list
```
Comment (stephaniehobson) [inline bedrock/firefox/views.py]: Where is the `d` coming from?

File: bedrock/firefox/views.py (hunk 3 above)
Comment (stephaniehobson) [reply bedrock/firefox/views.py]: Which is to say, did you intend to change the value of the `pre_download_locales` in this PR, I didn't think the experiment is related to this experiment. Maybe these lines have gotten out of sync through a rebase?

File: bedrock/firefox/views.py [hunk 5]
```diff
@@ -636,12 +636,15 @@ def download_thanks(request):
...
         variant = None
 
+    if variant in ['b', 'c'] and locale == 'en-US':
+        show_newsletter = False
+
     if variant == 'a' and locale in pre_download_locales:
```
Comment (stephaniehobson) [inline bedrock/firefox/views.py]: Hm, is the pre-download experiment using variant a as well? Should this experiment should use something different. x,y,z, maybe?

Comment (stephaniehobson) [reply bedrock/firefox/views.py]: Seems like `a` is the control for quite a few experiments so maybe this isn't a problem.

File: media/img/firefox/new/install/browser-illustration.svg [hunk 6]
```diff
@@ -0,0 +1 @@
+<svg height="249" width="360" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><g fill-rule="nonzero"><rect fill="#ededf0" height="42" rx="8" width="42" x="18" y="177"/><rect fill="#fff" height="249" rx="4" width="360"/><rect fill="#e8eaed" height="27" rx="4" width="360"/><rect fill="#fff" height="27" rx="4" width="63" x="14" y="10"/><g fill="#0060df"><path d="m41 148h22v41h-22z"/><path d="m24 179h56l-28 28z"/></g></g><path d="m325.5 11.5h5v5h-5z" stroke="#979797"/><path d="m307.5 13.5h6v1h-6zm35.879 2.121-.707-.707 4.242-4.242.707.707z" fill="#d8d8d8" stroke="#979797"/><path d="m343.379 11.379.707-.707 4.242 4.242-.707.707z" fill="#d8d8d8" stroke="#979797"/><path d="m0 216h360v1h-360zm116 6h-1v22h1z" fill="#d8d8d8"/><rect fill="#e8eaed" height="16" rx="8" width="230" x="77" y="37"/><rect fill="#c0c9d6" height="15" rx="3" width="72" x="32" y="226"/><rect fill="#c0c9d6" height="15" rx="3" width="15" x="9" y="226"/></g></svg>
```
Comment (stephaniehobson) [inline media/img/firefox/new/install/browser-illustration.svg]:
```suggestion
<svg height="249" width="360" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><g fill-rule="nonzero"><rect fill="#ededf0" height="42" rx="8" width="42" x="18" y="177"/><rect fill="#fff" height="249" rx="4" width="360"/><rect fill="#e8eaed" height="27" rx="4" width="360"/><rect fill="#fff" height="27" rx="4" width="63" x="14" y="10"/><g fill="#0060df"><path d="m41 148h22v41h-22z"/><path d="m24 179h56l-28 28z"/></g></g><path d="m325.5 11.5h5v5h-5z" stroke="#979797"/><path d="m307.5 13.5h6v1h-6zm35.879 2.121-.707-.707 4.242-4.242.707.707z" fill="#d8d8d8" stroke="#979797"/><path d="m343.379 11.379.707-.707 4.242 4.242-.707.707z" fill="#d8d8d8" stroke="#979797"/><path d="m0 216h360v1h-360zm116 6h-1v22h1z" fill="#d8d8d8"/><rect fill="#e8eaed" height="16" rx="8" width="230" x="77" y="37"/><rect fill="#c0c9d6" height="15" rx="3" width="72" x="32" y="226"/><rect fill="#c0c9d6" height="15" rx="3" width="15" x="9" y="226"/></g></svg>

```

File: media/js/firefox/new/install/experiment.js (hunk 1 above)
Comment (alexgibson) [reply media/js/firefox/new/install/experiment.js]: Crap, great catch Stephanie! Let me try and find some horrendous piece of code to fix this 😕

File: bedrock/firefox/views.py (hunk 3 above)
Comment (alexgibson) [reply bedrock/firefox/views.py]: Sorry, I mean to leave a comment for the reviewer here and totally forgot. I'm cleaning up some old experiment code here to make way for this new test to run (with the exception of `fr`, all pre-download experiments have ended).

File: bedrock/firefox/views.py (hunk 4 above)
Comment (alexgibson) [reply bedrock/firefox/views.py]: Leftover from when there we're 4 variants (one was scrapped), will update!

File: bedrock/firefox/views.py (hunk 5 above)
Comment (alexgibson) [reply bedrock/firefox/views.py]: Yep, this test is en-US only so shouldn't interfere with the pre-download locales.

**Comment by alexgibson** [comment]
> How can I inspect the "custom auth request"? is that in the request URL a request header?

For this particular experiment you can do the following:

1. Open Chrome on Windows (with DNT disabled)
2. Visit: https://bedrock-demo-agibson.oregon-b.moz.works/en-US/firefox/new/
3. Once redirected to a variation URL, wait for the page to load and then open Dev Tools.
4. Go to the Application tab and inspect cookies.
5. You should see a cookie called `moz-stub-attribution-code` with a base64 value like `c291cmNlPXd3dy5tb3ppbGxhLm9yZyZtZWRpdW09ZG93bmxvYWRfYnV0dG9uJmNhbXBhaWduPWRvd25sb2FkX3RoYW5rc19wYWdlJmNvbnRlbnQ9ZG93bmxvYWRfdGhhbmtzX2luc3RhbGxfZXhwZXJpbWVudC12LWM` Copy that value to the clip board.
6. Decode that base64 string (you can use a library, or something like https://www.base64decode.org/ for quickness).
7. Verify that the decoded value matches the utm params for that variation  e.g. `source=www.mozilla.org&medium=download_button&campaign=download_thanks_page&content=download_thanks_install_experiment-v-c`

**Comment by alexgibson** [comment]
@stephaniehobson thanks for the careful review. I added some terrible code that hopefully better detects Chrome. I've tested it in various VM's and it seems to work ok. I don't think we can't guarantee it will be entirely free of false positives, and is certainly not something I'd ever want us to use in fully featured code. Hopefully for now, this will do 😕

File: bedrock/firefox/views.py (hunk 3 above)
Comment (stephaniehobson) [reply bedrock/firefox/views.py]: 👍

File: media/js/firefox/new/install/experiment.js [hunk 7]
```diff
@@ -5,7 +5,36 @@
...
+        var isOpera = typeof window.opr !== 'undefined';
+        var isEdge = window.navigator.userAgent.indexOf('Edge') > -1;
+        var isYandex = window.navigator.userAgent.indexOf('YaBrowser') > -1;
+
+        // Limit the platform to Windows (i.e. desktop).
+        if (window.site.platform !== 'windows') {
+            return false;
+        }
+
+        // Attack of the clones!
```
Comment (stephaniehobson) [inline media/js/firefox/new/install/experiment.js]: 🤣

**Comment by stephaniehobson** [comment]
It's showing up in the anticipated browsers with my various Windows browsers now 👍  Looking at custom auth request now.

Needs a rebase.

**Comment by alexgibson** [comment]
Thanks @stephaniehobson rebased

**Pull request merged**